### DISCUSSION TITLE
Ressource for adapting 3D maxpool / convTranspose 3D kernels for MLX ?

### DISCUSSION THREAD

**Discussion by arnaudberenbaum** [ml-explore/mlx#1095] [Q&A] [answered]
Hello everyone, 
I'm working a lot with 3D medical imaging and I'm very interested in adapting 3D models for image processing for MLX.
From what I understood, MPS does not support 3D convolutions / ConvTranspose / Maxpool3D  and would need some new kernel.
Do you have any good ressource if I want to try to create these kernels? :) I've haven't got experience with low level GPU programming but i'm willing to learn 🙂
(just for information, I saw this implementation in pytorch : https://github.com/pytorch/pytorch/pull/116580 )
Thanks a lot !

**Comment by awni** [comment, marked as answer]
We just landed initial conv3d support in MLX: https://github.com/ml-explore/mlx/pull/993 so you can take a look at that.

**Reply by awni** [reply]
For Maxpool3D I think you should be able to do it with existing reductions + stride tricks rather than writing a kernel for it.

**Comment by arnaudberenbaum** [comment]
Ok thanks a lot, I will check this implementation 👍🏼

**Reply by francescopisu** [reply]
@arnaudberenbaum Any news on this?
